test/simulator/CMakeLists.txt

```diff
@@ -0,0 +1,185 @@
+  ${DBB-FIRMWARE-UI-SOURCES}
+  ${FIRMWARE-U2F-SOURCES}
+  ${DBB-FIRMWARE-USB-SOURCES}
+  )
+
+set(IGNORE_SOURCES
+    "src/screen.c"
+    "src/memory/nvmctrl.c"
+    "src/memory/smarteeprom.c"
+    "src/memory.mpu.c"
+    )
+
+# Exclude some files which depends on the hardware.
+foreach(SOURCEFILE ${IGNORE_SOURCES})
+    list(FILTER DBB-FILTERED-SOURCES EXCLUDE REGEX ".*/${SOURCEFILE}$")
+endforeach()
+
+message("FILTERED SOURCES: ${DBB-FILTERED-SOURCES}")
+
+add_library(sd-mock-simulator
+  STATIC
+  framework/mock_diskio.c
+)
+target_include_directories(
+  sd-mock-simulator
+  PUBLIC
+  $<TARGET_PROPERTY:fatfs,INTERFACE_INCLUDE_DIRECTORIES>
+)
+
+
+# We create a CMake "object library" to track all the compiled sources so that
+# they can be reused between a normal library and the manually crafted "merged"
+# library.
+# See https://cmake.org/cmake/help/v3.10/command/add_library.html#object-libraries
+add_library(bitbox_objects-simulator
+  OBJECT
+  ${DBB-FILTERED-SOURCES}
+  ${CTAES-SOURCES}
+  ${ETHEREUM-SOURCES}
+  framework/mock_cipher.c
+  framework/mock_memory.c
+  framework/mock_screen.c
+  framework/mock_smarteeprom.c
+  framework/mock_securechip.c
+)
+
+add_library(bitbox-simulator
+  STATIC
+  $<TARGET_OBJECTS:bitbox_objects-simulator>
+)
+
+# Here we create the "merged" library, which starts with the c-lib created by
+# cargo in the "bitbox02-rust-c" project. That project produces a c-lib without
+# mangled symbols so that it is easier to use from C. We then extend that
+# library with all the object files declared in "bitbox_objects-simulator", all the code
+# we want to be able to call from tests.
+# By linking to "bitbox_merged-simulator" in the rust unit tests we get access to all our
+# code. (That linkage is done by cargo using a "build.rs" script in the
+# bitbox02 rust project.)
+add_custom_command(
+  OUTPUT ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libbitbox_merged-simulator.a
+  DEPENDS c-unit-tests_rust_c bitbox_objects-simulator $<TARGET_OBJECTS:bitbox_objects-simulator>
+  COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_PROPERTY:c-unit-tests_rust_c,IMPORTED_LOCATION> ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libbitbox_merged-simulator.a
+  COMMAND ar q ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libbitbox_merged-simulator.a $<TARGET_OBJECTS:bitbox_objects-simulator>
+  VERBATIM
+  COMMAND_EXPAND_LISTS
+)
+
+add_custom_target(bitbox_merged-simulator DEPENDS ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libbitbox_merged-simulator.a)
+
+target_include_directories(
+  bitbox_objects-simulator
+  SYSTEM PUBLIC
+  ${SYSTEMINCLUDES}
+  ${CMAKE_SOURCE_DIR}/external
+  ${CMAKE_SOURCE_DIR}/external/ctaes
+  ${CMAKE_SOURCE_DIR}/external/fatfs/source
+  $<TARGET_PROPERTY:wallycore,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:secp256k1,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:fatfs,INTERFACE_INCLUDE_DIRECTORIES>
+)
+target_include_directories(
+  bitbox_objects-simulator
+  PUBLIC
+  ${INCLUDES}
+  ${CMAKE_CURRENT_SOURCE_DIR}/framework/includes
```

Comment:
we have a bit of a mess with the includes:

the libc-unit-tetsts library built in src/CMakeLists adds the  `unit-test/framework/includes`  directory (see https://github.com/digitalbitbox/bitbox02-firmware/blob/1bcb6dd934635a454d5ddb13524da569ec5ea1e3/src/CMakeLists.txt#L266), and the resulting lib is used in both the unit tests *and* the simulator.

the simulator then also has a copy of most/all of the same includes and adds it here.

untangling this could be annoying, let's postpone it? i think you can just use this instead

    ${CMAKE_CURRENT_SOURCE_DIR}/../unit-test/framework/includes

and delete the simulator/framework/includes folder.

Then all have the same includes, which might not be ideal, but at least consistent.

Reply:
simulator only uses 3 files from `framework/includes` folder, while unit-test uses 8. so actually simulator uses less stuff from there, but still we can postpone this for now since the solution would be untrivial. i'll change the `includes` path as `${CMAKE_CURRENT_SOURCE_DIR}/../unit-test/framework/includes` in the build of simulator.